mods/quick-hide-desktop-icons.wh.cpp

```diff
@@ -145,311 +141,342 @@
-        else if (wcscmp(hotkey, L"F2") == 0) g_settings.hotkey = VK_F2;
-        else if (wcscmp(hotkey, L"F3") == 0) g_settings.hotkey = VK_F3;
-        else if (wcscmp(hotkey, L"F4") == 0) g_settings.hotkey = VK_F4;
-        else if (wcscmp(hotkey, L"F5") == 0) g_settings.hotkey = VK_F5;
-        else if (wcscmp(hotkey, L"F6") == 0) g_settings.hotkey = VK_F6;
-        else if (wcscmp(hotkey, L"F7") == 0) g_settings.hotkey = VK_F7;
-        else if (wcscmp(hotkey, L"F8") == 0) g_settings.hotkey = VK_F8;
-        else if (wcscmp(hotkey, L"F9") == 0) g_settings.hotkey = VK_F9;
-        else if (wcscmp(hotkey, L"F10") == 0) g_settings.hotkey = VK_F10;
-        else if (wcscmp(hotkey, L"F11") == 0) g_settings.hotkey = VK_F11;
-        else if (wcscmp(hotkey, L"F12") == 0) g_settings.hotkey = VK_F12;
-        else g_settings.hotkey = (int)hotkey[0]; // Fallback to first char
-    }
+
+    g_settings.triggerMode = trigger ? (wcscmp(trigger, L"keyboard") == 0 ? 1 : wcscmp(trigger, L"both") == 0 ? 2 : 0) : 0;
+    g_settings.mouseButton = button && wcscmp(button, L"middle") == 0 ? 2 : 0;
+    g_settings.clickType = click && wcscmp(click, L"single") == 0 ? 0 : 1;
+    g_settings.hotkey = hotkey ? StringToVK(hotkey) : VK_F12;
+
+    Wh_FreeStringSetting(trigger);
+    Wh_FreeStringSetting(button);
+    Wh_FreeStringSetting(click);
     Wh_FreeStringSetting(hotkey);
 
-    // Load enableSound
-    g_settings.enableSound = Wh_GetIntSetting(L"enableSound");
+    g_settings.useCtrl = Wh_GetIntSetting(L"modifiers.ctrl") != 0;
+    g_settings.useAlt = Wh_GetIntSetting(L"modifiers.alt") != 0;
+    g_settings.useShift = Wh_GetIntSetting(L"modifiers.shift") != 0;
+    g_settings.enableSound = Wh_GetIntSetting(L"enableSound") != 0;
 
-    Wh_Log(L"Settings loaded: triggerMode=%d, mouseButton=%d, clickType=%d, hotkey=%d, ctrl=%d, alt=%d, shift=%d, sound=%d",
-           g_settings.triggerMode, g_settings.mouseButton, g_settings.clickType, g_settings.hotkey,
-           g_settings.useCtrl, g_settings.useAlt, g_settings.useShift, g_settings.enableSound);
+    QH_LOG(L"Settings loaded");
 }
 
-// Initialize mod
-BOOL Wh_ModInit() {
-    Wh_Log(L"Initializing");
+// Update hook state
+static void EnsureHookState() {
+    DWORD currentPid = GetCurrentProcessId();
+    DWORD shellPid = GetShellProcessIdCached();
+    bool isShellNow = shellPid && shellPid == currentPid;
+
+    if (isShellNow && !g_isShellProcess) {
+        g_isShellProcess = true;
+        if (!DispatchMessageW_Original) {
+            Wh_SetFunctionHook((void*)DispatchMessageW, (void*)DispatchMessageW_Hook, (void**)&DispatchMessageW_Original);
+            QH_LOG(L"Hook installed");
+        }
+    } else if (!isShellNow && g_isShellProcess) {
+        g_isShellProcess = false;
+        if (DispatchMessageW_Original) {
+            #ifdef Wh_RemoveFunctionHook
+            Wh_RemoveFunctionHook((void*)DispatchMessageW, (void*)DispatchMessageW_Hook);
+            #endif
```

Comment:
`Wh_RemoveFunctionHook` is not a macro, so this `#ifdef` block will never be compiled. In addition, `Wh_SetFunctionHook` and `Wh_RemoveFunctionHook` don't have any effect until `Wh_ApplyHookOperation` is called, unless hooks are set in `Wh_ModInit`. Generally, it's recommended to set the hooks once in `Wh_ModInit` and then use them when needed, or just call the original function.

Refer to the documentation for details:
https://github.com/ramensoftware/windhawk/wiki/Creating-a-new-mod#wh_setfunctionhook